core/src/main/java/com/graphhopper/routing/Path.java

```diff
@@ -703,7 +704,16 @@ private boolean surroundingStreetsAreSlowerByFactor(int baseNode, int prevNode,
                 double maxSurroundingSpeed = -1;
                 double tmpSpeed;
                 while (edgeIter.next()) {
-                    tmpSpeed = encoder.getSpeed(edgeIter.getFlags());
+                    if (encoder instanceof DataFlagEncoder) {
+                        tmpSpeed = ((DataFlagEncoder) encoder).getMaxspeed(edgeIter, 0, false);
```

Comment:
as we have the weighting in the Path class available, would it be somehow possible to use it here instead? E.g. edgeIter.getDistance/calcMillis?